### ISSUE TITLE
Fix S3459 FP: Unassigned members should be removed -> Does not recognize bindings from razor files / partial Blazor components

### ISSUE THREAD

**Issue by naefp** [SonarSource/sonar-dotnet#8538]
### Description

When declaring a private field in a razor.cs file, and bind it by ref in the related razor file, sonar marks it as not assigned. Sonar should check the related razor file for [S3459](https://sonarsource.github.io/rspec/#/rspec/S3459/csharp) as well.

### Repro steps

Create a Blazor component with two files:

MyBlazorComponent.razor
```html
<Modal @ref="@_myModalRef">
</Modal>
```

MyBlazorComponent.razor.cs
```c#
public partial class MyBlazorComponent
{
  private Modal _myModalRef; // Sonar marks this field as not assigned
}
```

### Expected behavior

Recognize that the field actually is being assigned by ref in razor file.

### Actual behavior

Mark every occurrence as false positive, or disable rule csharpsquid:S4487 for **/*.razor.cs completely.

Probably not using code in separate *razor.cs file but declare the private field in the razor file itself. Did not test that.

### Known workarounds

Please provide a description of any known workarounds.

### Related information

* C# 12
* Rider 2023.3.2
* .Net 8
* SonarLint plugin 10.2.1.77304 connected to SonarCloud
* Windows 11

**Comment by zsolt-kolbay-sonarsource** [comment]
Thank you for reporting this. Confirmed as False Positive.
